tests/child_thread/host/host.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/host.h>
+#include <openenclave/internal/calls.h>
+#include <openenclave/internal/error.h>
+#include <openenclave/internal/tests.h>
+#include <chrono>
+#include <cstdio>
+#include <cstdlib>
+#include <thread>
+
+#include "child_thread_u.h"
+
+#define ECALL_IN_CHILD_THREAD 0
+#define DESTROY_IN_CHILD_THREAD 1
+#define DESTROY_AND_CHECK_CHILD_THREAD 2
+
+bool multi_thread_flag = true;
+
+void stay_ocall(void)
+{
+    // sleep for 5 seconds.
+    std::this_thread::sleep_for(std::chrono::milliseconds(5000));
+}
+
+static void* child_thread_ecall(void* lpParam)
+{
+    multi_thread_flag = false;
+    oe_enclave_t* enclave = NULL;
+    enclave = (oe_enclave_t*)lpParam;
+    uint32_t magic;
+    oe_result_t result = get_magic_ecall(enclave, &magic);
+    OE_TEST(magic == 0x1234);
+    OE_TEST(result == OE_OK);
+    multi_thread_flag = true;
+    return NULL;
+}
+
+static void* child_thread_destroy(void* lpParam, oe_result_t expect_result)
+{
+    multi_thread_flag = false;
+    oe_enclave_t* enclave = NULL;
+    enclave = (oe_enclave_t*)lpParam;
+    oe_result_t result = oe_terminate_enclave(enclave);
+    fprintf(stdout, "oe_terminate_enclave returned: 0x%x\n", result);
+    OE_TEST(result == expect_result);
+    multi_thread_flag = true;
+    return NULL;
+}
+
+static void* child_thread_stay(void* lpParam)
+{
+    oe_enclave_t* enclave = NULL;
+    enclave = (oe_enclave_t*)lpParam;
+    int res;
+    oe_result_t result = stay_in_ocall_ecall(enclave, &res);
+    fprintf(stdout, "stay_in_ocall_ecall returned: 0x%x\n", result);
```

Comment:
I am not sure what the expected behavior is. If another thread terminates the enclave while this thread is in an ecall, do we expect this ecall to succeed?

Reply:
No. From my experience, the ecall should fail, and return some understandable error code.